zebrad/tests/acceptance.rs

```diff
@@ -1455,7 +1455,69 @@ async fn tracing_endpoint() -> Result<()> {
     Ok(())
 }
 
-// TODO: RPC endpoint and port conflict tests (#3165)
+#[tokio::test]
+async fn rpc_endpoint() -> Result<()> {
+    use hyper::{body::to_bytes, Body, Client, Method, Request};
+
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let endpoint = format!("127.0.0.1:{}", port);
+    let url = format!("http://{}", endpoint);
+
+    // Write a configuration that has RPC listen_addr set
+    let mut config = default_test_config()?;
+    config.rpc.listen_addr = Some(endpoint.parse().unwrap());
+
+    let dir = testdir()?.with_config(&mut config)?;
+    let child = dir.spawn_child(&["start"])?;
+
+    // Run `zebrad` for a few seconds before testing the endpoint
+    // Since we're an async function, we have to use a sleep future, not thread sleep.
+    tokio::time::sleep(LAUNCH_DELAY).await;
+
+    // Create an http client
+    let client = Client::new();
+
+    // Create a request to call `getinfo` RPC method
+    let req = Request::builder()
+        .method(Method::POST)
+        .uri(url)
+        .header("content-type", "application/json")
+        .body(Body::from(
+            r#"{"jsonrpc": "2.0", "method": "getinfo", "id":123}"#,
+        ))?;
+
+    // Make the call to the RPC endpoint
+    let res = client.request(req).await?;
+
+    // Test rpc endpoint response
+    assert!(res.status().is_success());
+
+    let body = to_bytes(res).await;
+    let (body, mut child) = child.kill_on_error(body)?;
+    let response_string = format!("{:?}", body);
+
+    // The `getinfo` RPC method returns the software version,
+    // this will always have `Zebra` as part of the response.
+    assert!(response_string.contains("Zebra"));
```

Comment:
Optional: We could parse JSON here for better test coverage.

Reply:
https://github.com/ZcashFoundation/zebra/pull/3624/commits/d83c0d4c2ccde65db7c4ab36debfe9513fd8c3f3